tap_asana/streams/stories.py

```diff
@@ -76,22 +82,55 @@ def get_objects(self):
             for project in self.call_api("projects", workspace=workspace["gid"]):
                 project_ids.append(project["gid"])
 
-        # iterate over all project ids and continue fetching
-        for project_id in project_ids:
-            for task in self.call_api("tasks", project=project_id):
-                task_gid = task.get("gid")
-                for story in Context.asana.client.stories.get_stories_for_task(
-                    task_gid=task_gid,
-                    opt_fields=opt_fields,
-                    timeout=self.request_timeout,
-                ):
-                    session_bookmark = self.get_updated_session_bookmark(
-                        session_bookmark, story[self.replication_key]
-                    )
-                    if self.is_bookmark_old(story[self.replication_key]):
-                        yield story
-
-        self.update_bookmark(session_bookmark)
+        project_ids_count = len(project_ids)
+
+        if project_ids_count == 0:
+            return
+
+        with ThreadPoolExecutor(max_workers=project_ids_count) as tasks_executor:
```

Comment:
#q what is the max length for `project_ids_count` and `task_gids_count` we expect to see?
#q what is the max reasonable thread pool size in the serverless environment ?

I wonder if we should additionally limit like `max_workers=min(32, project_ids_count)`.
I recall that asana pagination offers only 25 records at a time, so maybe it's not needed 🤔

Reply:
There might be a lot of tasks per project, so your point regarding possible memory consumption makes sense

Reply:
The asana pagination page size is 100

Reply:
<img width="931" alt="Screenshot 2023-08-26 at 17 05 12" src="https://github.com/railsware/tap-asana/assets/10974851/c3f412cf-0871-47f8-9b58-0cf37bd64401">

Reply:
What about Asana API limit? When we tested it there was a warning about max requests and this limit is 10 requests. Is there some info about it in the docs? I would not create as many threads as we have tasks (or projects). We need to find the optimized number. Maybe `max_workers=min(100, project_ids_count)` or something like this. As I remember our cloud functions have only 2 cores.